### PR TITLE
drivers: optee: Introduce basic OP-TEE driver to Zephyr

### PR DESCRIPTION
**Pull request by oleksiimoisieiev**
This introduces the TEE subsystem definition and OP-TEE driver basic implementation with the unit tests
This is preliminary implementation which implements tee syscalls except for shm_register and shm_unregister calls.
DID NOT TESTED on the real hardware, only unit tests.

### CODE REVIEW

**Comment by xakep-amatop** [comment]
you are adding a new driver and device abstraction it would be nice to add some documentation to zephyr/doc

File: tests/drivers/tee/tee/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+tests:
+  drivers.tee.tee:
+    build_only: false
+    tags: tee
+    timeout: 500
+    integration_platforms:
+      - native_posix_64
+    platform_allow: native_posix_64
```
Comment (xakep-amatop) [inline tests/drivers/tee/tee/testcase.yaml:8]: should you add R-Car gen3 & gen4 boards ?

**Comment by oleksiimoisieiev** [comment]
> you are adding a new driver and device abstraction it would be nice to add some documentation to zephyr/doc

+1. I will add docs.

File: drivers/tee/optee/Kconfig [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2023 Epam Systems
+# SPDX-License-Identifier: Apache-2.0
+
+config OPTEE
+	bool "Enable optee driver in system"
+	depends on TEE
```
Comment (xakep-amatop) [inline drivers/tee/optee/Kconfig:6]: `depends on ARM || ARM64 || RISCV`

File: tests/drivers/tee/tee/testcase.yaml (hunk 1 above)
Comment (oleksiimoisieiev) [reply tests/drivers/tee/tee/testcase.yaml:8]: I think not. I see no reason of running those test on the target boards. The idea is that this are the unit tests, that doesn't have direct connection to the optee. For integration testing we have zephyr-optee-test app.
I'm preparing for MR to xen-troops so all fixes will include to the new MR.

File: drivers/tee/optee/Kconfig (hunk 2 above)
Comment (oleksiimoisieiev) [reply drivers/tee/optee/Kconfig:6]: +

**Comment by oleksiimoisieiev** [comment]
> you are adding a new driver and device abstraction it would be nice to add some documentation to zephyr/doc

Tee.h has doxygen notes across the file. 
I'm preparing for MR to xen-troops so all fixes will include to the new MR.

**Comment by oleksiimoisieiev** [comment]
Closing this PR as the new one for zephyr on xen-troops should be created.

**Pull request closed without merging**